src/coreclr/hosts/corebundle/TPA.cpp

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <stdio.h>
+#include <windows.h>
+#include "TPA.h"
+#include "palclr.h"
+
+bool TPA::HasFile(_In_z_ char* fileNameWithoutExtension)
```

Comment:
Why do we need the TPA at all in this case?
Also: Was this code copied from somewhere or is it all new?

Reply:
Technically, the TPA builder is not necessary for the self-contained single-file apps with no additional dependencies. 

Given the fact that we load R2R images via copy, I kept the TPA construction logic so that we can compare the startup cost when:
1) R2R assemblies are loaded from bundle (no TPA)
2) R2R assemblies are loaded from files located alongside the bundled-app (partial single-file, requires TPA).

The TPA logic was copied from `coreconsole` with some minor adjustments (uses char* instead of wchar_t* etc).